File: app/src/main/java/com/kirchhoff/example/githubclient/api/interceptor/AuthorizationInterceptor.java
package com.kirchhoff.example.githubclient.api.interceptor;

import android.support.annotation.NonNull;

import com.kirchhoff.example.githubclient.GitHubApplication;
import com.kirchhoff.example.githubclient.utils.TextUtils;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * @author Kirchhoff-
 */

public class AuthorizationInterceptor implements Interceptor {

    private final String token;

    private AuthorizationInterceptor() {
        token = GitHubApplication.getAppComponent().keyValueStorage().getToken();
    }

    @NonNull
    public static Interceptor create() {
        return new AuthorizationInterceptor();
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        if (TextUtils.isEmpty(token)) {
            return chain.proceed(chain.request());
        }

        Request request = chain.request().newBuilder()
                .addHeader("Authorization", String.format("%s %s", "token", token)).build();
        return chain.proceed(request);
    }
}

File: app/src/main/java/com/kirchhoff/example/githubclient/api/interceptor/AuthorizationInterceptor.kt
package com.kirchhoff.example.githubclient.api.interceptor

import com.kirchhoff.example.githubclient.Injection
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

/**
 * @author Kirchhoff-
 */

class AuthorizationInterceptor private constructor() : Interceptor {

    private val token: String = Injection.provideKeyValueStorage().token

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        if (token.isEmpty()) {
            return chain.proceed(chain.request())
        }

        val request = chain.request().newBuilder()
                .addHeader("Authorization", String.format("%s %s", "token", token)).build()
        return chain.proceed(request)
    }

    companion object {

        fun create(): Interceptor {
            return AuthorizationInterceptor()
        }
    }
}